### PR TITLE
ExPlat: Remove unused tests and deprecate calypso-abtests

### PR DESCRIPTION
**Pull request by jessie-ross**
**This PR commences deprecating the `calypso-abtests` library.**

Part of the work on pbxNRc-sZ-p2 and under the issue: 495-gh-Automattic/experimentation-platform

* Adds deprecation messages
* Uses JSDoc to mark the API as deprecated
* **Removes these tests without any code referencing them:**
   * `springSale30PercentOff`
   * `skipThemesSelectionModal`
   * `builderReferralStatsNudge`
   * `newSiteGutenbergOnboarding`
   
_If you are wondering why your review is being requested it is because you are on the git blame for one or more of the above tests._

#### Testing instructions

No significant code changes so no testing should be necessary.

### CODE REVIEW

File: client/lib/abtest/README.md [hunk 1]
```diff
@@ -1,3 +1,9 @@
+
+# This library is deprecated!
+
+*Please consider ExPlat for your next A/B experiment.*
+See pbxNRc-sZ-p2 or visit #a8c-experiments and ask around!
```
Comment (yanirs) [inline client/lib/abtest/README.md]: It's probably better to link to `/client/components/experiment/readme.md` and add the P2 and Slack links there.

File: client/lib/abtest/active-tests.js [hunk 2]
```diff
@@ -4,6 +4,11 @@
  * paths here (e.g. `import * from '../../lib/`)
  */
 
+/**************************************************************************************************/
+/* This library is deprecated! Please consider ExPlat for your next A/B experiment.               */
+/* See pbxNRc-sZ-p2 or visit #a8c-experiments and ask around!
```
Comment (yanirs) [inline client/lib/abtest/active-tests.js]: It'd also be good to change these links to point to the new component readme, given that it's open source and seen by people outside a8c (not that they'd be running A/B tests, but still...).

File: client/desktop/window-handlers/failed-to-load/index.js [hunk 3]
```diff
@@ -71,24 +71,25 @@ function failedToLoadError( mainWindow ) {
...
-				if ( isErrorPage( event.sender ) ) {
-					failedToLoadError( mainWindow );
-				} else {
-					log.error(
-						'Failed to load from server, showing fallback page: code=' +
-							errorCode +
-							' ' +
-							errorDescription
-					);
+	mainWindow.webContents.on( 'did-fail-load', async function (
```
Comment (yanirs) [inline client/desktop/window-handlers/failed-to-load/index.js]: Seems like the latest commit brought in a bunch of unrelated formatting changes... :thinking:

File: client/my-sites/purchases/site-level-purchases-error-boundary.tsx [hunk 4]
```diff
@@ -7,7 +7,9 @@ import styled from '@emotion/styled';
 
 const debug = debugFactory( 'calypso:site-level-purchases-error-boundary' );
 
-export default class SiteLevelPurchasesErrorBoundary extends React.Component< SiteLevelPurchasesErrorBoundaryProps > {
+export default class SiteLevelPurchasesErrorBoundary extends React.Component<
```
Comment (sirbrillig) [inline client/my-sites/purchases/site-level-purchases-error-boundary.tsx]: Is this change safe? I seem to remember reading somewhere that this would break some things and was fixed in a recent version of Prettier. But maybe I'm misremembering.

File: client/lib/cart/store/cart-synchronizer.js [hunk 5]
```diff
@@ -142,18 +142,17 @@ CartSynchronizer.prototype._processQueuedChanges = function () {
...
-		this._cartKey,
-		preprocessCartForServer( this._latestValue ),
-		function ( error, newValue ) {
-			if ( error ) {
-				callback( error );
-				return;
-			}
-
-			callback( null, preprocessCartFromServer( newValue ) );
+	this._wpcom.setCart( this._cartKey, preprocessCartForServer( this._latestValue ), function (
```
Comment (sirbrillig) [inline client/lib/cart/store/cart-synchronizer.js]: AFAICT this doesn't change anything in this file, so I assume it's just a formatting change. Was this intentional as part of this PR?

File: packages/composite-checkout/src/components/checkout-steps.tsx [hunk 6]
```diff
@@ -879,7 +880,9 @@ const StepNumberCompleted = styled( StepNumber )`
 	// prettier-ignore
 	@media all and (-ms-high-contrast:none), (-ms-high-contrast:active) {
 		backface-visibility: visible;
-		z-index: ${ ( props ) => ( props.isComplete ? '1' : '0' ) };
+		z-index: ${ (
+		props
```
Comment (sirbrillig) [inline packages/composite-checkout/src/components/checkout-steps.tsx]: I'm not sure this is necessary either, and I think it might actually cause issues. What was the reason for this change?

**Comment by jessie-ross** [comment]
Oops I messed up the merge, let's try this again.

**Review by yanirs — APPROVED** [review]
LGTM
